src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -186,23 +207,41 @@ def _try_to_breakdown_plan(self, root: TreeNode) -> List[LogicalPlan]:
             return [root]
 
         plans = []
-        # TODO: SNOW-1617634 Have a one pass algorithm to find the valid node for partitioning
+        final_partition_breakdown_summary = {}
         while complexity_score > self.complexity_score_upper_bound:
-            child = self._find_node_to_breakdown(root)
+            child, validity_statistics = self._find_node_to_breakdown(root)
             if child is None:
+                final_partition_breakdown_summary = {
+                    k.value: validity_statistics.get(k, 0)
+                    for k in [
+                        CompilationStageTelemetryField.INVALID_DUE_TO_SCORE_ABOVE_UPPER_BOUND,
```

Comment:
Why not have a enum class specific for InValidBreadownNodeReason so that we can group the reasons together, and then you don't need those "INVALID_DUE_TO" prefix, which could also make the code easier to understand.

Reply:
Also if you have an enum class, we can simply iterate over the enum class by doing
" for reason in InValidBreadownNodeReason"